File: dp_primer/dp_primer_recursive_formula_step3.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] x = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] a = new int[1000];
        int q = int.Parse( ReadLine() );

        SetOut( new System.IO.StreamWriter( OpenStandardOutput() ) { AutoFlush = false } );
        a[0] = x[0];
        foreach( int i in Range( 1, 999 ) ) {
            a[i] = i % 2 == 0 ? a[i - 1] + x[1] : a[i - 1] + x[2];
        }
        foreach( int _ in Range( 0, q ) ) {
            WriteLine( a[int.Parse( ReadLine() ) - 1] );
        }
        Out.Flush();
        return;
    }
}

File: dp_primer/dp_primer_recursive_formula_step3.py
# 実行時間 10ms
import sys

rl = sys.stdin.readline
x, d1, d2 = list(map(int, rl().split()))
q = int(rl())
a = [x]

for i in range(1, 1000):
    a.append(a[-1] + d1 if i % 2 == 0 else a[-1] + d2)
for _ in [0] * q:
    print(a[int(rl()) - 1])
